modules/core/src/smithy4s/schema/SchemaPartition.scala

```diff
@@ -181,11 +196,13 @@ object SchemaPartition {
           label: String,
           instance: Schema[A],
           get: S => Option[A]
-      ): SchemaPartition[S] = SchemaPartition.NoMatch()
+      ): SchemaPartition[S] =
+        // "single" only makes sense on required fields
+        SchemaPartition.NoMatch()
     }
 
   private def codingError: Nothing =
-    sys.error("Coding error: this should not happen on encoding side")
+    sys.error("Coding error: this should not happen on the encoding side")
```

Comment:
**issue:** btw this doesn't only appear in encoding I think, `codingError` is used once on the decoding side (`from` in line 187)